tools/code_format/check_format_test_helper.py

```diff
@@ -232,6 +232,12 @@ def runChecks():
       "Registry::InjectFactory instead.")
   errors += checkUnfixableError("strerror.cc",
                                 "Don't use strerror; use Envoy::errorDetails instead")
+  errors += checkUnfixableError(
+      "std_unordered_map.cc", "Don't use std::unordered_map; use absl::flat_hash_map instead " +
```

Comment:
We seem to be preferring the node_hash_map, not flat_hash_map.

Reply:
Ah, we put `flat_hash_map` as the first choice since it is recommended for new code and `node_hash_map` is the stabler drop-in replacement for `std::unordered_map`, we can flip the order or remove one option if you like

Reply:
I prefer we emphasize flat_hash_map. The node_hash_map makes certain additional guarantees that shouldn't be needed in new code.

Reply:
Recommending flat_hash_map seems fine.  It may be worth pointing to docs in cases where flat_hash_map doesn't work.

Do we intend to do a second pass to change node_hash_map use to flat_hash_map where doing so is safe?

Reply:
We certainly should, but the windows port is unblocked by merging this current iteration. It sounds like a perfect Help Wanted issue for a contributor with the round tuits to spend. We have a couple months ahead of the more convoluted fixes to work through before looping back on good tickets like this.

Reply:
(Actually, a Help Wanted and Tech Debt tag on that follow up issue)